packages/notebook/src/browser/notebook-cell-resource-resolver.ts

```diff
@@ -0,0 +1,74 @@
+// *****************************************************************************
+// Copyright (C) 2023 TypeFox and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import { Resource, ResourceReadOptions, ResourceResolver, ResourceVersion, URI } from '@theia/core';
+import { inject, injectable } from '@theia/core/shared/inversify';
+import { Emitter } from '@theia/core/shared/vscode-languageserver-protocol';
+import { CellDto, CellUri } from '../common';
+import { NotebookService } from './service/notebook-service';
+import { NotebookModel } from './view-model/notebook-model';
+
+class NotebookCellResource implements Resource {
+
+    private readonly didChangeContentsEmitter = new Emitter<void>();
```

Comment:
Suggestion: We leave nothing `private` in Theia (at least outside of the plugin-host).

```suggestion
    protected readonly didChangeContentsEmitter = new Emitter<void>();
```